.github/workflows/build.yml

```diff
@@ -26,18 +26,15 @@ jobs:
       - name: Checkout
         uses: actions/checkout@v4
 
-      - name: Gradle JDK
-        uses: actions/setup-java@v4
-        with:
-          distribution: temurin
-          java-version: 17
-          cache: gradle
-
       - name: Gradle build, test and check
-        uses: gradle/gradle-build-action@v3
+        uses: ./.github/workflows/gradle-goal
         with:
-          arguments: check
+          command: "./gradlew check"
```

Comment:
[minor] maybe it would be worth to keep the `./gradlew` part handled in the `gradle-goal` workflow and only provide the gradle arguments here.

Reply:
I can do that

Reply:
Wait, if I do so, then the gradle-goal will convert in `*nix` based. So windows won't be supported.

Let's keep it for now

Reply:
Good for me, that's definitely something we can improve later.